daemon/src/source_pair_import.rs

```diff
@@ -232,21 +256,29 @@ impl<'a> SourcePairImport<'a> {
                     ),
                 ));
                 let asset_data = &asset.asset_data;
-                let serialized_asset =
-                    SerializedAsset::create(asset_data.as_ref(), CompressionType::None, scratch_buf)?;
+                ctx.begin_serialize_asset(asset.id);
+                let serialized_asset = SerializedAsset::create(
+                    asset_data.as_ref(),
+                    CompressionType::None,
+                    scratch_buf,
+                )?;
+                let mut deps = ctx.end_serialize_asset(asset.id);
                 let build_pipeline = metadata
                     .assets
                     .iter()
                     .find(|a| a.id == asset.id)
                     .and_then(|m| m.build_pipeline);
+                for load_dep in asset.load_deps {
+                    deps.insert(load_dep);
+                }
                 imported_assets.push({
                     ImportedAsset {
                         asset_hash: utils::calc_asset_hash(&asset.id, import_hash),
                         metadata: AssetMetadata {
                             id: asset.id,
                             search_tags: asset.search_tags,
                             build_deps: asset.build_deps,
-                            load_deps: asset.load_deps,
+                            load_deps: deps.into_iter().collect(),
```

Comment:
looks like a no-op?

Reply:
Yeah sorry it should've been a HashSet, fixed